src/js/controllers/correspondentDevices.js

```diff
@@ -56,31 +58,69 @@ angular.module('copayApp.controllers').controller('correspondentDevicesControlle
 				$scope.error = err;
 				return;
 			}
+
+			correspondentListService.readNotRemovableDevices(function(err, arrNotRemovableDeviceAddresses) {
+
+				// add a new property indicating whether the device can be removed or not
+				
+				var length = ab.length;
+				for (var i = 0; i < length; i++) {
+ 				 	corrDev = ab[i];
+
+				 	corrDevAddr = corrDev.device_address;
+					
+				 	var ix = arrNotRemovableDeviceAddresses.indexOf(corrDevAddr);
+					
+					// device is removable when not in list
+				 	corrDev.removable = (ix == -1);
+				}
+			});
+		
 			$scope.list = ab;
 			$scope.$digest();
 		});
 	};
 	
+	$scope.hideRemoveButton = function(removable){
+		return $scope.hideRemove || !removable;
+	}
 
-	$scope.remove = function(addr) {
-		throw Error("unimplemented");
-		$scope.error = null;
-		$timeout(function() {
-		  correspondentListService.remove(addr, function(err, ab) {
-			if (err) {
-			  $scope.error = err;
-			  return;
+	$scope.remove = function(device_address) {
+
+		// check to be safe
+		correspondentListService.deviceCanBeRemoved(device_address, {
+			ifOk: function(){
```

Comment:
When a device cannot be removed, this situation doesn't seem like an "error".
Why not just return a boolean, wrapping the callbacks into an object seems unnecessary